src/modules/CurrencyModule.cpp

```diff
@@ -915,20 +917,123 @@ CurrencyModule::CurrencyModule(UmikoBot* client) : Module("currency", true), m_c
 			return;
 		}
 
+		QString thiefName = UmikoBot::Instance().GetName(channel.guildId(), authorId);
+		QString victimName = UmikoBot::Instance().GetName(channel.guildId(), victimId);
+
+		auto& victimCurrency = guildList[channel.guildId()][getUserIndex(channel.guildId(), victimId)];
+		auto& authorCurrency = guildList[channel.guildId()][getUserIndex(channel.guildId(), authorId)];
+
+		EventModule* eventModule = static_cast<EventModule*>(UmikoBot::Instance().GetModuleByName("event"));
+		auto& eventConfig = eventModule->getEventServerData(channel.guildId());
+
+		if (eventConfig.eventHighRiskHighRewardRunning)
```

Comment:
(This comment applies to a lot of the steal function, i haven't quoted it all for compactness sake).

I feel like it would be better to calculate the variables once and then have the steal logic execute to avoid code duplication. Instead of copy-pasting the steal code in these `if` statements a bunch of times, we should instead create local variables for the success chance, victim bonus, thief penalty, etc. Then have the steal logic run with those variables, so you don't need to have the steal logic copied many times here.

Reply:
Yes, I have done this (kind of in the new PR)